### PR TITLE
feat(Azure): Adds support for Azure Service Bus worker queues as an Azure alternative to Redis and AWS SQS. (LL-37)

### PR DESCRIPTION
**Pull request by cbishopvelti**
Please remember to request a review from someone when this PR is ready for a code review.

#### Which issue does this close?
Closes https://learningpool.atlassian.net/browse/LL-37.

#### Describe your changes providing screenshots of UI changes if necessary.

Adds support for Azure service bus.

### CODE REVIEW

File: lib/services/queue/serviceBus/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      logger.error(err);
+      await brokeredMessage.deadLetter({
+        deadLetterErrorDescription: JSON.stringify(err, null, 2),
+        deadletterReson: 'Message was not valid JSON'
+      });
+    }
+    try {
+      const renewLock = async () => {
+        let visibilityTimeoutUpdated;
+        while (visibilityTimeoutUpdated > 0) {
```
Comment (h-kanazawa) [inline lib/services/queue/serviceBus/index.js]: `visibilityTimeoutUpdated` seems always to be undefined;

File: lib/services/queue/serviceBus/index.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    done(err);
+    throw err;
+  }
+
+  if (!retryDelay) {
+    await sender.send({
+      body: stringPayload
+    });
+  } else if (retryDelay) {
+    await sender.schedualmessage(moment().add(retryDelay, 'seconds').toDate(), {
```
Comment (h-kanazawa) [inline lib/services/queue/serviceBus/index.js]: Is this method name `schedualmessage` correct?

File: lib/services/queue/serviceBus/index.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  } catch (err) {
+    done(err);
+    throw err;
+  }
+
+  if (!retryDelay) {
+    await sender.send({
+      body: stringPayload
+    });
+  } else if (retryDelay) {
```
Comment (h-kanazawa) [inline lib/services/queue/serviceBus/index.js]:
```suggestion
  } else {
```

File: lib/services/queue/serviceBus/index.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      if (isString(brokeredMessage.body)) {
+        payload = JSON.parse(brokeredMessage.body);
+      } else {
+        payload = brokeredMessage.body;
+      }
+    } catch (err) {
+      logger.error(err);
+      await brokeredMessage.deadLetter({
+        deadLetterErrorDescription: JSON.stringify(err, null, 2),
+        deadletterReson: 'Message was not valid JSON'
```
Comment (h-kanazawa) [inline lib/services/queue/serviceBus/index.js]: typo?
```suggestion
        deadletterReason: 'Message was not valid JSON'
```

File: .env.example [hunk 5]
```diff
@@ -181,6 +181,9 @@ QUEUE_NAMESPACE=DEV
 #PUBSUB_GOOGLE_CLOUD_PROJECT_ID=
 #PUBSUB_GOOGLE_CLOUD_SUBSCRIPTION_NAME=ll
 
+# Azure service bus endpoint, required if QUEUE_PROVIDER=SERVICE_BUS
+# SERVICE_BUS_ENDPOINT='sb://[namespace].servicebus.windows.net/;SharedAccessKeyName=[keyName];SharedAccessKey=[key]';
```
Comment (ryansmith94) [inline .env.example]:
```suggestion
# SERVICE_BUS_ENDPOINT='Endpoint=sb://[namespace].servicebus.windows.net/;SharedAccessKeyName=[keyName];SharedAccessKey=[key]'
```

**Pull request merged**